event-sourcing/subscription-manager/src/main/java/uk/gov/justice/services/event/sourcing/subscription/CdiEventSourceFactory.java

```diff
@@ -0,0 +1,22 @@
+package uk.gov.justice.services.event.sourcing.subscription;
+
+import uk.gov.justice.services.eventsourcing.source.core.EventSource;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.spi.BeanManager;
+import javax.enterprise.inject.spi.CDI;
+import javax.enterprise.inject.spi.Unmanaged;
+
+@ApplicationScoped
+public class CdiEventSourceFactory {
+
+    @SuppressWarnings("unchecked")
+    public EventSource create(final Class<? extends EventSource> eventSourceClass) {
```

Comment:
This code should just move into the producer - the producer **is** the factory. Squashing this will simplify this code change hugely; you then just need to test that the producer gives you a DefaultEventSource or throws an exception.

Reply:
done